File: src/AltinnCore/Common/Models/RepoStatus.cs
using System.Collections.Generic;
using AltinnCore.Common.Enums;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace AltinnCore.Common.Models
{
    /// <summary>
    /// This class define the Repository status
    /// </summary>
    public class RepoStatus
    {
        /// <summary>
        /// The number of commits behind the 
        /// </summary>
        public int? BehindBy { get; set; }

        /// <summary>
        /// The number of commits before the remote
        /// </summary>
        public int? AheadBy { get; set; }

        /// <summary>
        /// List over files that has changed
        /// </summary>
        public List<RepositoryContent> ContentStatus { get; set; }

        /// <summary>
        /// Defines the status of the repository
        /// </summary>
        [JsonConverter(typeof(StringEnumConverter))]
        public RepositoryStatus RepositoryStatus { get; set; }

        /// <summary>
        /// Defines if there is any merge conflicts
        /// </summary>
        public bool HasMergeConflict { get; set;  }
    }
}

File: src/Designer/frontend/packages/shared/src/components/GiteaHeader/VersionControlButtons/utils/repoStatus.ts
import { type RepoStatus } from 'app-shared/types/RepoStatus';

export const hasRepoMergeConflict = (repoStatus?: Partial<RepoStatus>): boolean =>
  Boolean(repoStatus?.hasMergeConflict || repoStatus?.repositoryStatus === 'MergeConflict');

export const hasCheckoutConflict = (repoStatus?: Partial<RepoStatus>): boolean =>
  repoStatus?.repositoryStatus === 'CheckoutConflict';

export const toMergeConflictRepoStatus = (repoStatus?: Partial<RepoStatus>): RepoStatus => ({
  ...repoStatus,
  hasMergeConflict: true,
  repositoryStatus: 'MergeConflict',
  contentStatus: repoStatus?.contentStatus ?? [],
  aheadBy: repoStatus?.aheadBy ?? 0,
  behindBy: repoStatus?.behindBy ?? 0,
});
